test/clients.go

```diff
@@ -38,8 +38,15 @@ type KubeClient struct {
 }
 
 // NewSpoofingClient returns a spoofing client to make requests
-func NewSpoofingClient(client *KubeClient, logf logging.FormatLogger, domain string, resolvable bool, opts ...spoof.TransportOption) (*spoof.SpoofingClient, error) {
-	return spoof.New(client.Kube, logf, domain, resolvable, Flags.IngressEndpoint, opts...)
+func NewSpoofingClient(client *KubeClient, logf logging.FormatLogger, domain string, resolvable bool,
+	opts ...spoof.TransportOption) (*spoof.SpoofingClient, error) {
+	return NewSpoofingClientWithOptions(client, logf, domain, resolvable, nil, opts...)
+}
+
+// NewSpoofingClientWithOptions returns a spoofing client to make requests with spoofing options
+func NewSpoofingClientWithOptions(client *KubeClient, logf logging.FormatLogger, domain string, resolvable bool,
+	clientOpts *spoof.SpoofingClientOptions, opts ...spoof.TransportOption) (*spoof.SpoofingClient, error) {
```

Comment:
```suggestion
	clientOpts spoof.SpoofingClientOptions, opts ...spoof.TransportOption) (*spoof.SpoofingClient, error) {
```
no need to pass by pointer. Also saves you hassle later.

Reply:
It's easier to test for empty on pointer though, so that it can set default. But I guess it's OK to have this function to set default. Changing it to pass by value then.